docs/_integration-with-existing-apps-kotlin.md

```diff
@@ -4,94 +4,81 @@ import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import con
 
 The keys to integrating React Native components into your Android application are to:
 
-1. Set up React Native dependencies and directory structure.
-2. Develop your React Native components in JavaScript.
-3. Add a `ReactRootView` to your Android app. This view will serve as the container for your React Native component.
-4. Start the React Native server and run your native application.
-5. Verify that the React Native aspect of your application works as expected.
+1. Set up the correct directory structure
+2. Install the necessary NPM dependencies
+3. Adding React Native to your Gradle configuration
+4. Writing the TypeScript code for your first React Native screen
+5. Integrate React Native with your Android code using a ReactActivity
+6. Testing your integration by running the bundler and seeing your app in action
+
+## Using the Community Template
+
+While you follow this guide, we suggest you to use the [React Native Community Template](https://github.com/react-native-community/template/) as reference. The template contains a **minimal Android app** and will help you understanding how to integrate React Native into an existing Android app.
 
 ## Prerequisites
 
 Follow the guide on [setting up your development environment](set-up-your-environment) and using [React Native without a framework](getting-started-without-a-framework) to configure your development environment for building React Native apps for Android.
+This guide also assumes you're familiar with the basics of Android development such as creating Activities and editing the `AndroidManifest.xml` file.
 
-### 1. Set up directory structure
+## 1. Set up directory structure
 
-To ensure a smooth experience, create a new folder for your integrated React Native project, then copy your existing Android project to an `/android` subfolder.
+To ensure a smooth experience, create a new folder for your integrated React Native project, then **copy your existing Android project** to the `/android` subfolder.
 
-### 2. Install JavaScript dependencies
+## 2. Install NPM dependencies
 
-Go to the root directory for your project and create a new `package.json` file with the following contents:
+Go to the root directory and run the following command:
 
 ```
-{
-  "name": "MyReactNativeApp",
-  "version": "0.0.1",
-  "private": true,
-  "scripts": {
-    "start": "yarn react-native start"
-  }
-}
+wget https://raw.githubusercontent.com/react-native-community/template/refs/heads/0.75-stable/template/package.json
```

Comment:
this hides a lot of complexity. Wouldn't it be better to keep the instructions to add the `package.json` and to install react-native and react as they were before?

Reply:
Discussed offline.

The problem is that users will have to install as a bare minimum at least also:

```
    "@babel/core": "^7.20.0",
    "@babel/preset-env": "^7.20.0",
    "@babel/runtime": "^7.20.0",
    "@react-native/babel-preset": "0.75.3",
    "@react-native/metro-config": "0.75.3",
    "@react-native/typescript-config": "0.75.3",
    "@types/react": "^18.2.6",
    "typescript": "5.0.4"
```

As otherwise Metro and Codegen won't work. Typescript is also needed down the line.